docs/advanced.md

```diff
@@ -622,3 +622,541 @@ If you want to access those values from within your `Fastfile` use
+if UI.interactive?
+  name = UI.input "Please enter your name: "
+  is_correct = UI.confirm "Is this correct? "
+  choice = UI.select "Please choose from the following list:", %w{alpha beta gamma}
+  password = UI.password "Please enter your password: "
+end
+```
+
+|method|description|
+|------|-----------|
+|interactive?|Indicates whether interactive input is possible|
+|input|Prompt the user for string input|
+|confirm|Ask the user a binary question|
+|select|Prompt the user to select from a list of options|
+|password|Prompt the user for a password (masks output)|
+
+## Invoking shell commands
+
+If your action needs to run a shell command, there are several methods. You can
+easily determine the exit status of the command and capture all terminal output
+from the command.
+
+### Using Kernel::system
+
+Use the Ruby `system` method call to invoke a command string. This does not
+redirect stdin, stdout or stderr, so output formatting will be unaffected. It executes
+the command in a subshell.
+
+```ruby
+system "cat fastlane/Fastfile"
+```
+
+Upon command completion, the method returns. The `$?` global variable will
+indicate the exit status of the command.
+
+```ruby
+system "cat fastlane/Fastfile"
+UI.user_error! "Could not execute command" unless $?.exitstatus == 0
+```
+
+If the command to be executed is not found,
+`$?.exitstatus` will be nonzero.
+
+### Using backticks
+
+To capture the output of a command, enclose the command in backticks:
+
+```ruby
+pod_cmd = `which pod`
+UI.important "'pod' command not found" if pod_cmd.empty?
+```
+
+Because you are capturing stdout, the command output will not appear at
+the terminal unless you log it using `UI`. Formatting may be lost when
+capturing command output. The entire output to stdout will be captured after the
+command returns. Output to stderr is not captured or redirected. The `$?`
+global variable will indicate the completion status of the command.
+
+If the command to be executed is not found, `Errno::ENOENT` is raised.
+
+### Using the sh action
+
+You can also use the built-in `sh` Fastlane action:
+
+```ruby
+other_action.sh "pwd"
+```
+
+Within a Fastfile, just call `sh`, e.g.: `sh "pwd"`. This provides consistent
+logging of command output. All output to stdout and stderr is logged via `UI`.
+
+To be notified when an error occurs, use the `error_callback` parameter:
+
+```ruby
+success = true
+other_action.sh("pwd", error_callback: ->(result) { success = false })
+UI.user_error "Command failed" unless success
+```
+
+The `result` argument to the `error_callback` is the entire string output
+of the command. The exit status of the command will be available in `$?`.
+
+If the command to be executed is not found, `Errno::ENOENT` is raised.
+
+### Escaping in shell commands
+
+Use `shellwords` to escape arguments to shell commands.
+
+```Ruby
+`git commit -aqm#{Shellwords.escape commit_message}`
+```
+
+```Ruby
+system "cat #{path.shellescape}"
+```
+
+## Calling other actions
```

Comment:
@ohayon I tried to summarize here the discussion from https://github.com/fastlane/fastlane/pull/10675. It's deliberately a bit vague. Any thoughts on what to say here?